service_configuration_lib/spark_config.py

```diff
@@ -647,6 +647,10 @@ def _calculate_resources(
         'spark.executor.instances': str(executor_instances),
         'spark.task.cpus': str(task_cpus),
     })
+    if 'spark.cores.max' in user_spark_opts:
+        core_max = int(user_spark_opts.get('spark.cores.max', 0))
+        required_cores = executor_instances * executor_cores
+        user_spark_opts['spark.cores.max'] = str(max(core_max, required_cores))
```

Comment:
Why do we need to take max(given_max_cores, calculated_max_cores)?
Should we just override given_max_cores to calculated_max_cores, and WARN user to overridden value and ask them to remove from the given configs?

Reply:
Sure, let me change to overwriting the value. For warning we already have one here: https://github.com/Yelp/service_configuration_lib/pull/107/files/5ce41f5b3deb428b35e19594d2f80ba0d4e7e0df#diff-7e57e6df6168e4083ee9a3b454025d48cb1f68c100d45dd92cf9056010cd5220R480

Reply:
Nit: https://github.com/Yelp/service_configuration_lib/pull/107/files/5ce41f5b3deb428b35e19594d2f80ba0d4e7e0df#diff-7e57e6df6168e4083ee9a3b454025d48cb1f68c100d45dd92cf9056010cd5220R479

Replace warning message to: 
log.warning(
            'spark.cores.max is DEPRECATED. Replace with '
            'spark.executor.instances={executor_instances} in --spark-args and in your service code as "spark.executor.instances * spark.executor.cores" if used.',
        )

Reply:
I updated the message and change the logic a little bit in the latest commit to always show it, regardless of whether `spark.executor.instances` is set or not.